File: java/pe/src/ktn/pe/Pe12.java
package ktn.pe;

public class Pe12 implements Pe {
    private int over = 500;

    public int pe12() {
        return pe12(over);
    }

    public int pe12(int over) {
        int t = 1;
        int i = 2;
        int count;
        while (true) {
            count = PeUtils.countDivisors(t);
            if (count >= over) {
                break;
            }
            t += i;
            ++i;
        }
        
        return t;
    }
    
    @Override
    public void run() {
        solve();
    }

    @Override
    public int getProblemNumber() {
        return 12;
    }

    @Override
    public void setArgs(String[] args) {
        if (args == null) { return; }
        if (args.length > 0) {
            try {
                over = Integer.parseInt(args[0]);
            } catch (NumberFormatException e) {
                //
            }
        }
    }

    @Override
    public void solve() {
        System.out.println(PeUtils.format(this, pe12()));
    }

}

File: kt/pe/src/main/kotlin/ktn/pe/Pe12.kt
package ktn.pe

class Pe12 : Pe {
    private var _over = 500

    override val problemNumber: Int
        get() = 12

    fun pe12(over: Int = _over): Int {
        var t = 1
        var i = 2
        var count: Int
        while (true) {
            count = PeUtils.countDivisors(t.toLong())
            if (count >= over) {
                break
            }
            t += i
            ++i
        }

        return t
    }

    override fun run() {
        solve()
    }

    override fun setArgs(args: Array<String>) {
        if (args.isNotEmpty()) {
            _over = args[0].toInt()
        }
    }

    override fun solve() {
        System.out.println(PeUtils.format(this, pe12()))
    }

}
